src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] storage = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        if (size == storage.length) {
+            increaseCapacity();
+        }
+        storage[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        indexValidation(index, false);
 
+        if (index == size) {
+            add(value);
+            return;
+        }
+
+        if (size == storage.length) {
+            increaseCapacity();
+        }
+        System.arraycopy(storage, index, storage, index + 1, size - index);
+        storage[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            return;
```

Comment:
Better to throw runtime exception, cause the user couldn't find the mistake if he has a null list